### ISSUE TITLE
Ubuntu armhf build test failure

### ISSUE THREAD

**Issue by jbicha** [google/brotli#649]
The build tests for brotli 1.0.3 fail on armhf on Ubuntu 18.04 but they passed with 1.0.2.

It does build on Debian's armhf, but I'm told that Ubuntu's builder is a bit different. If I understand correctly, Ubuntu's armhf builder is actually an arm64 machine that emulates armhf. It was suggested that this might be an alignment bug.

Full build log at
https://launchpad.net/ubuntu/+source/brotli/1.0.3-1/+build/14449807

The Debian build logs are at
https://buildd.debian.org/status/package.php?p=brotli
(Just click `Installed`)

Build log excerpt
-----------------------
```
test__test_decompress__xyzzy (tests.decompressor_test.TestDecompressor) ... ok
test__test_decompress__zeros (tests.decompressor_test.TestDecompressor) ... ok
test__test_compress_lgwin_10_quality_11_10x10y (tests.compress_test.TestCompress) 
     ... Bus error (core dumped)
```

**Comment by eustas** [comment]
Going to investigate ASAP (on Thursday)

**Comment by thinred** [comment]
I suspect it's the same as https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=892895.

**Comment by eustas** [comment]
Plain clang sanitizer does not catch the problem. Trying `QEMU` + `chroot` for emulating armhf...

**Comment by eustas** [comment]
Failed to emulate with `QEMU` + `chroot`, but managed to do it with docker.
Still, emulation seems to be weak - wasn't able to reproduce the bug.

Attached the binary I got by cross-compiling to `arm32v7` under `aarch64`. I would appreciate if you test it in your environment, e.g. `brotli -Zfkw 10 some-compressible-file`. Also it would be nice if you share the executable which fails, so I would be able to play with it. Thanks.

[brotli.gz](https://github.com/google/brotli/files/1825168/brotli.gz)

**Comment by eustas** [comment]
Got RPi3 into my hands. Unfortunately, it thinks that it is `BCM2709` (`armv7`) instead of `BCM2837` (`armv8` / `aarch64`). Is there an easy way to fix it?

**Comment by eustas** [comment]
Running openSUSE (`aarch64`)  on RPi3. Looking for the ways to simulate what `sbuild` does, i.e. cross-compile and run (32-bit) `armhf` binary.

**Comment by eustas** [comment]
Was able to reproduce. Fails in `platform.h:200`. `BrotliUnalignedWrite64`. Going to dig why it is triggered in `armhf` under `aarch64` build.

**Comment by eustas** [comment]
It seems that problem is not with any unaligned memory access, but specific to 64-bit read/write in 32-bit mode. Going to see if there is elegant workaround for this.

**Comment by eustas** [comment]
Should be fixed with #656. Going to release v1.0.4 soon

**Comment by eustas** [comment]
Changes released in v1.0.4
